tests/e2e/configurer/configurer.go

```diff
@@ -0,0 +1,513 @@
+	dockerImages "github.com/osmosis-labs/osmosis/v7/tests/e2e/configurer/docker"
+	"github.com/osmosis-labs/osmosis/v7/tests/e2e/util"
+)
+
+type Configurer interface {
+	ConfigureChains() error
+
+	ClearResources() error
+
+	GetChainConfig(chainIndex int) ChainConfig
+
+	RunSetup() error
+
+	RunValidators() error
+
+	RunIBC() error
+
+	SendIBC(srcChain *chain.Chain, dstChain *chain.Chain, recipient string, token sdk.Coin)
+
+	CreatePool(chainId string, valIdx int, poolFile string)
+}
+
+type BaseConfigurer struct {
+	chainConfigs   []*ChainConfig
+	dockerImages   *dockerImages.ImageConfig
+	dockerPool     *dockertest.Pool
+	dockerNetwork  *dockertest.Network
+	valResources   map[string][]*dockertest.Resource
+	hermesResource *dockertest.Resource
+	setupTests     setupFn
+	t              *testing.T
+}
+
+type status struct {
+	LatestHeight string `json:"latest_block_height"`
+}
+
+type syncInfo struct {
+	SyncInfo status `json:"SyncInfo"`
+}
+
+const (
+	// osmosis version being upgraded to (folder must exist here https://github.com/osmosis-labs/osmosis/tree/main/app/upgrades)
+	UpgradeVersion = "v9"
+	// estimated number of blocks it takes to submit for a proposal
+	PropSubmitBlocks float32 = 10
+	// estimated number of blocks it takes to deposit for a proposal
+	PropDepositBlocks float32 = 10
+	// number of blocks it takes to vote for a single validator to vote for a proposal
+	PropVoteBlocks float32 = 1.2
+	// number of blocks used as a calculation buffer
+	PropBufferBlocks float32 = 5
+	// max retries for json unmarshalling
+	MaxRetries = 60
+)
+
+var (
+	// whatever number of validator configs get posted here are how many validators that will spawn on chain A and B respectively
+	validatorConfigsChainA = []*chain.ValidatorConfig{
+		{
+			Pruning:            "default",
+			PruningKeepRecent:  "0",
+			PruningInterval:    "0",
+			SnapshotInterval:   1500,
+			SnapshotKeepRecent: 2,
+		},
+		{
+			Pruning:            "nothing",
+			PruningKeepRecent:  "0",
+			PruningInterval:    "0",
+			SnapshotInterval:   1500,
+			SnapshotKeepRecent: 2,
+		},
+		{
+			Pruning:            "custom",
+			PruningKeepRecent:  "10000",
+			PruningInterval:    "13",
+			SnapshotInterval:   1500,
+			SnapshotKeepRecent: 2,
+		},
+		{
+			Pruning:            "everything",
+			PruningKeepRecent:  "0",
+			PruningInterval:    "0",
+			SnapshotInterval:   0,
+			SnapshotKeepRecent: 0,
+		},
+	}
+	validatorConfigsChainB = []*chain.ValidatorConfig{
+		{
+			Pruning:            "default",
+			PruningKeepRecent:  "0",
+			PruningInterval:    "0",
+			SnapshotInterval:   1500,
+			SnapshotKeepRecent: 2,
+		},
+		{
+			Pruning:            "nothing",
+			PruningKeepRecent:  "0",
+			PruningInterval:    "0",
+			SnapshotInterval:   1500,
+			SnapshotKeepRecent: 2,
+		},
+		{
+			Pruning:            "custom",
+			PruningKeepRecent:  "10000",
+			PruningInterval:    "13",
+			SnapshotInterval:   1500,
+			SnapshotKeepRecent: 2,
+		},
+	}
+)
```

Comment:
thoughts on moving this to its own file?

Reply:
I moved the constructor and validator configs into `factory.go` while the constants to `constants.go`

I think it makes sense to keep the validator configs together with the factory constructor for the ease of referencing / initialization. Since `factory.go` is not meant to have more methods or functions, this looks clean and concise. Let me know what you think